GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker is flaky

# Summary

The test `io.camunda.zeebe.it.gateway.GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker` failed once but then passed in the same CI run.

## How often does the test fail?

At least once:
- [here](https://github.com/camunda/zeebe/actions/runs/4254244991/jobs/7400218352)

## Does it block your work?

I was notified as Medic that the build failed on `main`.

## Do we suspect that it is a real failure?

A 503 Service Unavailable likely indicates that the gateway was not fully started when the test was run.

# Failures

<details><summary>Example assertion failure</summary>
<pre>
Warning:  Flakes: 
Warning:  io.camunda.zeebe.it.gateway.GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker
Error:    Run 1: GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker:112 1 expectation failed.
Expected status code <503> but was <200>.
</pre>
</details>

**Hypotheses**

It may be that the CI was slow and didn't have enough time to start the Gateway 🤷 

**Logs**

<details><summary>Logs</summary>
<pre>
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 23.601 s <<< FAILURE! - in io.camunda.zeebe.it.gateway.GatewayHealthProbeIntegrationTest
Error:  io.camunda.zeebe.it.gateway.GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker  Time elapsed: 5.341 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <503> but was <200>.

	at java.base/jdk.internal.reflect.GeneratedConstructorAccessor133.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)
	at io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)
	at jdk.internal.reflect.GeneratedMethodAccessor341.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)
	at io.restassured.specification.ResponseSpecification$statusCode$0.callCurrent(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:143)
	at io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:89)
	at io.camunda.zeebe.it.gateway.GatewayHealthProbeIntegrationTest.shouldReportLivenessDownIfNotConnectedToBroker(GatewayHealthProbeIntegrationTest.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:234)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:228)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:131)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
</pre>
</details>

--- a/util/src/main/java/io/camunda/zeebe/util/health/DelayedHealthIndicator.java
+++ b/util/src/main/java/io/camunda/zeebe/util/health/DelayedHealthIndicator.java
@@ -46,9 +46,9 @@ public class DelayedHealthIndicator implements HealthIndicator {
   protected DelayedHealthIndicator(
       final HealthIndicator originalHealthIndicator,
       final Duration maxDowntime,
-      Supplier<Long> clock) {
+      final Supplier<Long> clock) {
     if (requireNonNull(maxDowntime).toMillis() < 0) {
-      throw new IllegalArgumentException("maxDonwtime must be >= 0");
+      throw new IllegalArgumentException("maxDowntime must be >= 0");
     }
     this.originalHealthIndicator = requireNonNull(originalHealthIndicator);
     this.maxDowntime = maxDowntime;
@@ -60,7 +60,7 @@ public class DelayedHealthIndicator implements HealthIndicator {
 
   public DelayedHealthIndicator(
       final HealthIndicator originalHealthIndicator, final Duration maxDowntime) {
-    this(originalHealthIndicator, maxDowntime, () -> System.currentTimeMillis());
+    this(originalHealthIndicator, maxDowntime, System::currentTimeMillis);
   }
 
   @Scheduled(fixedDelay = 5000)
@@ -92,7 +92,7 @@ public class DelayedHealthIndicator implements HealthIndicator {
     return responseBuilder.withDetails(createDetails(now)).build();
   }
 
-  private Map<String, Object> createDetails(long referenceTime) {
+  private Map<String, Object> createDetails(final long referenceTime) {
     final var result = new HashMap<>(staticDetails);
 
     if (lastHealthStatus != null) {

--- a/util/src/main/java/io/camunda/zeebe/util/health/DelayedHealthIndicator.java
+++ b/util/src/main/java/io/camunda/zeebe/util/health/DelayedHealthIndicator.java
@@ -78,7 +78,7 @@ public class DelayedHealthIndicator implements HealthIndicator {
     final long now = clock.get();
 
     if (lastHealthStatus == null) { // was never checked
-      responseBuilder = Health.unknown();
+      responseBuilder = Health.down();
     } else {
       if (lastTimeUp == null) { // was never up
         responseBuilder = Health.status(lastHealthStatus.getStatus());

--- a/util/src/test/java/io/camunda/zeebe/util/health/DelayedHealthIndicatorTest.java
+++ b/util/src/test/java/io/camunda/zeebe/util/health/DelayedHealthIndicatorTest.java
@@ -47,7 +47,7 @@ public class DelayedHealthIndicatorTest {
   }
 
   @Test
-  public void shouldReportUnknownHealthStatusIfAskedBeforeDelegateHealthIndicatorWasCalled() {
+  public void shouldReportDownHealthStatusIfAskedBeforeDelegateHealthIndicatorWasCalled() {
     // given
     final var sutDelayedHealthIndicator =
         new DelayedHealthIndicator(mockHealthIndicator, TEST_MAX_DOWNTIME);
@@ -57,7 +57,7 @@ public class DelayedHealthIndicatorTest {
 
     // then
     Assertions.assertThat(actualHealth).isNotNull();
-    Assertions.assertThat(actualHealth.getStatus()).isEqualTo(Status.UNKNOWN);
+    Assertions.assertThat(actualHealth.getStatus()).isEqualTo(Status.DOWN);
   }
 
   @Test

